test/functional/api/highlight_spec.lua

```diff
@@ -0,0 +1,54 @@
+
+local helpers = require('test.functional.helpers')(after_each)
+local clear, nvim = helpers.clear, helpers.nvim
+local Screen = require('test.functional.ui.screen')
+local eq, eval = helpers.eq, helpers.eval
+local command = helpers.command
+local ok = helpers.ok
+local meths = helpers.meths
+
+
+describe('highlight api', function()
+
+  before_each(function()
+    clear('--cmd', 'set termguicolors')
```

Comment:
does it require RGB? why?

Reply:
I tested the most useful configuration there. I can add without termguicolors as well.

Reply:
Won't it be useful for non-rgb?

Reply:
The only non-RGB UIs I can think of are terminal based and even those tend to disappear as always more terminals support RGB. I hope that neovim-based UIs won't limit themselves to 16 colors. Anyway that should work even with non-RGB. I can add the tests.